src/maxminddb.c

```diff
@@ -1220,7 +1223,7 @@ static int lookup_path_in_array(const char *path_elem,
 
     int saved_errno = errno;
     errno = 0;
-    int array_index = strtol(path_elem, &first_invalid, 10);
+    long array_index = strtol(path_elem, &first_invalid, 10);
```

Comment:
Should this be `unsigned long`? I ask because on L#1241 it casts it to unsigned ala `(unsigned long)array_index >= size`

Reply:
Ideally, but `strtol()` returns a `long`:

```
maxminddb.c:1226:33: warning: implicit conversion changes signedness: 'long' to 'unsigned long' [-Wsign-conversion]                                                     
    unsigned long array_index = strtol(path_elem, &first_invalid, 10);                                                                                                 
                  ~~~~~~~~~~~   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

We check it after for being `>= 0` so casting it after checking that is safe I believe. But I don't think it'd be safe to do it immediately.